search/searcher/search_numeric_range.go

```diff
@@ -63,6 +75,34 @@ func NewNumericRangeSearcher(indexReader index.IndexReader,
 		true)
 }
 
+func filterCandidateTerms(indexReader index.IndexReader,
+	terms [][]byte, field string) (rv [][]byte, err error) {
+	rv = make([][]byte, 0)
+	fieldDict, err := indexReader.FieldDictRange(field, terms[0], terms[len(terms)-1])
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
```

Comment:
Here, on the one hand defer near the resource creation feels idiomatic go'ish.

On the other hand, I guess the rest of the function doesn't exit early, so moving this to the end can skip a defer